### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by yaqwsx**

### CODE REVIEW

File: softwareComponents/snake_reconfig/Snake_algorithms.cpp [hunk 1]
```diff
@@ -228,8 +228,8 @@ std::vector<Configuration> reconfigThroughSnake(const Configuration& from, const
 
 std::vector<Configuration> aerateConfig(const Configuration& init) {
     SimpleNextGen simpleGen{};
-    SpaceGridScore gridScore(init.getIDs().size());
-    auto moduleCount = init.getModules().size();
+    SpaceGridScore gridScore(unsigned(init.getIDs().size()));
+    auto moduleCount = unsigned(init.getModules().size());
```
[the remark this answers is not in the document]
Comment (yaqwsx) [reply softwareComponents/snake_reconfig/Snake_algorithms.cpp:232]: I can add them. But based on what algorithm this implements, assuming 2^31 modules in the input configuration is a bit unrealistic.

File: softwareComponents/snake_reconfig/Snake_algorithms.cpp [hunk 2]
```diff
@@ -238,15 +240,17 @@ std::vector<Configuration> aerateConfig(const Configuration& init) {
 std::vector<Configuration> aerateFromRoot(const Configuration& init) {
     SmartBisimpleOnlyRotGen smartGen{};
     AwayFromRootScore rootScore{};
-    unsigned limit = 3 * init.getModules().size();
+    assert(fits<unsigned>(init.getModules().size()));
+    unsigned limit = 3 * unsigned(init.getModules().size());
```
Comment (Ondraceq) [inline softwareComponents/snake_reconfig/Snake_algorithms.cpp:244]: Use `auto` at the left side? - I don't mind either, but it's inconsistent

File: softwareComponents/snake_reconfig/Snake_algorithms.cpp [hunk 3]
```diff
@@ -318,7 +320,8 @@ std::vector<Configuration> makeEdgeSpace(const Configuration& init, ID subroot1,
 
     SmartBisimpleOnlyRotGen smartGen{};
     EdgeSpaceScore edgeScore(realMass, subtrees);
-    unsigned limit = 2 * init.getModules().size();
+    assert(fits<unsigned>(init.getModules().size()));
+    unsigned limit = 2 * unsigned(init.getModules().size());
```
Comment (Ondraceq) [inline softwareComponents/snake_reconfig/Snake_algorithms.cpp:324]: Again inconsistent

File: softwareComponents/snake_reconfig/Snake_algorithms.cpp [hunk 4]
```diff
@@ -422,17 +426,16 @@ void computeSubtreeSizes(const Configuration& config, std::unordered_map<ID, uns
...
+    for (size_t bagIndex = 0; bagIndex < moduleCount; ++bagIndex) {
         auto currId = bag[bagIndex];
         for (const auto& optEdge : spannSucc.at(currId)) {
             if (!optEdge.has_value())
                 continue;
             bag.emplace_back(optEdge.value().id2());
         }
     }
-    for (int i = moduleCount - 1; i >= 0; --i) {
+    for (int i = int(moduleCount) - 1; i >= 0; --i) {
```
Comment (Ondraceq) [inline softwareComponents/snake_reconfig/Snake_algorithms.cpp:438]: Use fits?

File: softwareComponents/snake_reconfig/Snake_structs.cpp [hunk 5]
```diff
@@ -93,7 +93,7 @@ double DistFromConnScore::distFromConn(const Configuration& config, const Edge&
 
 SpaceGridScore::SpaceGridScore(const Configuration& init)
 : _configuration(&init)
-, _module_count(init.getIDs().size())
+, _module_count(unsigned(init.getIDs().size()))
```
Comment (Ondraceq) [inline softwareComponents/snake_reconfig/Snake_structs.cpp:96]: Maybe add fits to the constructor body

**Review by Ondraceq — APPROVED** [review]
I wrote to all the places that I feel could be improved, but I'll leave to you if you want to follow it or not

**Pull request merged**